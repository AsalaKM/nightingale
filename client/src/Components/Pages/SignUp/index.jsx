import React, { Component } from "react";
import swal from "sweetalert";
import axios from "axios";

import validate from "./validation";

import Title from "./../../Common/Title";
import AppTitleImage from "./../../../assets/header.png";
import AppLogoImage from "./../../../assets/logo.png";
import Input from "./../../Common/Input";
import Button from "./../../Common/Button";

import {
  AppTitle,
  AppLogo,
  Form,
  SignUpWrapper,
  SignUpText,  
  SignUpLink,
  ErrorBox
} from "./index.style";

export default class SignUp extends Component {
  state = {
    name: {
      value: "",
      errMsg: ""
    },
    username: {
      value: "",
      errMsg: ""
    },
    password: {
      value: "",
      errMsg: ""
    },
    rePassword:{
      value: "",
      errMsg: ""
    },
  };

  handleSubmit = event => {
    const handleChangeState = this.props.handleChangeState;
    event.preventDefault();
    
    // axios
    //   .post("/api/user/signup", this.state)
    //   .then(res => {})
    //   .catch(err => {});
  };

  // validate(event.target.name, event.target.value).then(res=>{
  //   console.log(res);
  // }).catch(err=> {
  //   console.log(err.details[0].message);
    
  // })
  
  handleBlur = (event) => {
    console.log(2222222222);
    const key = event.target.name
    const value = event.target.value
      validate(key, value).then(res=>{
      }).catch(err=> {
        const newState = {
          ...this.state,
          [key]: {
            ...this.state[key],
            errMsg: err.details[0].message
          }
        }
        console.log(newState);
        
    this.setState(newState);
  })
  }
  handleChange = event => {
    const key = event.target.name
    const value = event.target.value

const newState = {
  ...this.state,
  [key]: {
    errMsg: "",
    value
  }
}
console.log(newState);
    this.setState(newState);
  };

  render() {
    return (
      <SignUpWrapper>
        <AppTitle src={AppTitleImage} />
        <AppLogo img src={AppLogoImage} />
        <Form onSubmit={this.handleSubmit}>
          <Title value="Sign Up" />
          <Input
            placeholder="Name"
            name="name"
            value={this.state.name.value}
            onChange={this.handleChange}
            onBlur={this.handleBlur}
          />
          {
            this.state.name.errMsg && 
          <ErrorBox>
            <p>{this.state.name.errMsg}</p>
          </ErrorBox>
          }
          <Input
            placeholder="Username"
            name="username"
            value={this.state.username.value}
            onChange={this.handleChange}
            onBlur={this.handleBlur}
          />
                    {
            this.state.username.errMsg && 
          <ErrorBox>
            <p>{this.state.username.errMsg}</p>
          </ErrorBox>
          }
          <Input
            placeholder="Password"
            name="password"
            value={this.state.password.value}
            type="password"
            onChange={this.handleChange}
            onBlur={this.handleBlur}
          />
                    {
            this.state.password.errMsg && 
          <ErrorBox>
            <p>{this.state.password.errMsg}</p>
          </ErrorBox>
          }
          <Input
            placeholder="Re-Password"
            name="rePassword"
            value={this.state.rePassword.value}
            type="password"
            onChange={this.handleChange}
          />
          <Button value="Sign Up" />
          <SignUpText>
            Have an account? <SignUpLink to="/login">Login</SignUpLink>
          </SignUpText>
        </Form>
      </SignUpWrapper>
    );
  }
}
