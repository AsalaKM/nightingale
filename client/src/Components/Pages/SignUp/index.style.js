import styled from "styled-components";
import { Link } from "react-router-dom";

export const SignUpWrapper = styled.div`
  width: 100%;
  padding: 2rem 0.5rem 5rem 0.5rem;
  min-height: 100vh;
  text-align: center;
  @media (min-width: 768px) {
    padding: 0rem;
    box-sizing: border-box;
  }
`;

export const MainLogo = styled.img`
  width: 55%;
  margin: 0 auto;
  @media (min-width: 700px) {
    display: block;
    width: 13%;
    margin: 0 auto;
  }
`;

export const Form = styled.form`
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-item: center;
  width: 30%;
  padding: 25px 0;
  box-shadow: 0px 0px 6px #ccc;
  margin-top: 25px;
  border-radius: 8px;
  max-width: 500px;

  button {
    max-width: 100px;
    margin: 12px auto;
  }
  h1 {
    border: none;
    margin-bottom: 1rem;
    font-weight: bold;
  }
  @media (max-width: 768px) {
    width: 85%;
  }
`;

export const InputWrapper = styled.div`
  position: relative;
  margin: 0 auto;
`;

export const Input = styled.input`
  outline: none;
  border: solid 0.5px #c8c7cc;
  margin: 1rem 0;
  display: block;
  height: 35px;
  padding: 0 1rem;
  border-radius: 100px;
  outline-color: orange;
  font-size: 16px;
  width: 100%;
  max-width: 400px;
`;

export const SignUpText = styled.p`
  text-align: center;
`;

export const SignUpLink = styled(Link)`
  display: block;
  color: #925de5;
  :hover {
    display: block;
    color: #925de5;
  }
  ,
  :active {
    display: block;
    color: #925de5;
  }
  ,
  :visited {
    display: block;
    color: #925de5;
  }
  ,
  :focus {
    display: block;
    color: #925de5;
  }
`;

export const ErrorBox = styled.div`
  position: absolute;
  min-width: 95%;
  background-color: #cccccc38;
  margin: 0 auto;
  text-align: center;
  border-radius: 4px;
  padding: 5px 10px;
  color: red;
  transform: translate(-50%, -50%);
  top: 102%;
  z-index: 1;
  left: 50%;
  font-size: 12px;

  &:before {
    display: block;
    position: absolute;
    left: 10%;
    top: -5px;
    content: "";
    width: 0;
    height: 0;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-bottom: 5px solid #cccccc38;
  }
`;
