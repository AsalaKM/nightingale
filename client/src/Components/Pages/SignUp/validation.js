const Joi = require("joi");

const validate = (key, value) => {
  const schemas = {
      name: Joi.string()
        .alphanum()
        .min(3)
        .max(15)
        .required()
        .label("Name"),
      username: Joi.string()
        .min(3)
        .max(10)
        .required()
        .label("Username"),

      password: Joi.string()
        .min(3)
        .required()
        .label("Password"),
  }
  console.log(key, "key ", value ," value", typeof value ,"type value");
  
  return Joi.validate(value, schemas[key]);
};
export default validate;