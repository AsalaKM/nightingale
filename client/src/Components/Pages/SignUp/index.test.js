import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import SignUp from "./index.jsx";

describe("SignUp must match the snapshots", () => {
  it("renders correctly", () => {
    const context = {};
    const tree = renderer
      .create(
        <StaticRouter context={context}>
          <SignUp />
        </StaticRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
