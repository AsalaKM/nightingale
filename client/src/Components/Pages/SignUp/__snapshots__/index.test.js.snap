// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp must match the snapshots renders correctly 1`] = `
<div
  className="sc-bxivhb hqlsQb"
>
  <img
    className="sc-ifAKCX djVFOV"
    src="header.png"
  />
  <img
    className="sc-EHOje lbgmuu"
    src="logo.png"
  />
  <form
    className="sc-bZQynM ggoioL"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdVaJa khUSez"
    >
      <h1
        className="sc-bwzfXH dtARpq"
      >
        Sign Up
      </h1>
    </div>
    <div
      className="sc-gzVnrw ekbgNh"
    >
      <input
        autoFocus={true}
        className="sc-htoDjs ekjgyT"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
      
    </div>
    <div
      className="sc-gzVnrw ekbgNh"
    >
      <input
        className="sc-htoDjs ekjgyT"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Username"
        value=""
      />
      
    </div>
    <div
      className="sc-gzVnrw ekbgNh"
    >
      <input
        className="sc-htoDjs ekjgyT"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      
    </div>
    <div
      className="sc-gzVnrw ekbgNh"
    >
      <input
        className="sc-htoDjs ekjgyT"
        name="rePassword"
        onChange={[Function]}
        placeholder="Re-Password"
        type="password"
        value=""
      />
      
    </div>
    <button
      className="sc-htpNat dSGjhB"
      style={
        Object {
          "background": undefined,
        }
      }
      type="submit"
    >
      Sign Up
    </button>
    <p
      className="sc-dnqmqq fXmtAI"
    >
      Have an account? 
      <a
        className="sc-iwsKbI hTGocj"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </p>
  </form>
</div>
`;
